### ISSUE TITLE
Track if user has logged in previously

### ISSUE THREAD

**Issue by rbbeeston** [boostorg/website-v2#1294]
When going to the site, only a "Join" label appears for logging in. However if a person has logged in previously and has been logged out they must go to join and then sign up to login.

What we should have happen is on successful login add a cookie or local storage flag that the user has logged in before. then that flag is checked on page load and replaces the "join" link with a "login" link that takes them directly to the login page.

**Comment by vinniefalco** [comment]
We need to be careful here. The login system is... a massive headache.

**Comment by sdarwin** [comment]
I just commented on the other ticket, where there was a misunderstanding.  As far as can be seen, things are working as expected, there is no bug, and no need to add cookies or something.    Social Auth is provided by a plugin, that is well maintained, and correct, for the most part.

I am able to log into an existing an account, even if I make a mistake and use the Join page instead of the log-in page. Both ways, it's fine.   You were switching from Google to Github. If you login first with Google, you can change it.

**Comment by rbbeeston** [comment]
@sdarwin that's not what this issue is in reference to. For example:
1. User logs in with google
2. user signs out (or login times out) and leaves site.
3. next time the user returns to the site, they must click the "join" button
4. once going to the "signup" page, they need to click the link to log in.
5. they then need to click the google button to login.

What I am proposing is to have a flag (yes the plugin may already have one, and that's fine to use that) to recognize if a user has signed in previously, and in that case, instead of having a "Join" link in the header, it would have a "login" link to bypass the signup pages before getting to a login page.

**Comment by sdarwin** [comment]
OK.
twitter, github, stackexchange all show **both** buttons:  "log in" "sign up".  
better to not add more cookies.

**Comment by vinniefalco** [comment]
No, I don't think we should do that Rob. If you are using a public terminal, or if you are using your computer at your house and a roommate has access to it, you might not want to leave behind information which suggests that you previously logged into a particular site.

**Comment by rbbeeston** [comment]
well, we already do that.  should we get rid this one?
![Screenshot 2024-09-26 at 4 43 22 PM](https://github.com/user-attachments/assets/1344c0e3-42bb-44c0-a4f4-fbe85b0832f3)

**Comment by vinniefalco** [comment]
Usually there is a "Remember me on this computer" checkbox

**Comment by rbbeeston** [comment]
I haven't seen one here, so did you want me to make up a ticket for "Remember me on this computer" and otherwise hide the last login info?

**Comment by vinniefalco** [comment]
we should have an issue, yes.

**Comment by rbbeeston** [comment]
Dave has dug into this a bit more, and a software update to the plugin and the email sending piece seemed to resolve most of it.

However, the "last login" notification still exists, and it was coded on our end. as there isn't a great place to put the checkmark, and for the sake of security, the ability to turn it on and off will be moved to the account settings. 

The only question that remains is whether or not that should be opt-in or opt-out. I favor opt-in for better default security.

**Comment by sdarwin** [comment]
@daveoconnor in the above screenshot that Rob posted it showed a "Last Log In".   How was that accomplished before, in the absence of a (new) cookie?  What was the technique there?
